SIPS/sip-keeper_synths.md

```diff
@@ -0,0 +1,72 @@
+The current version of the iSynths (inverse priced Synths) have price limits to protect the network, when these price limits are hit the Synth is frozen and must be restarted. Currently the protocolDAO is the only address that can call these functions. This results in delays and frustration for users. By making these functions public and paying a fee in SNX for anyone who resets an iSynth we can signficantly improve the user experience.
+
+## Abstract
+This SIP will change the iSynths purge and update pricing functions to be public and incentivised. The first person to call each function will be paid SNX as an incentive.
+
+## Motivation
+While the iSynths are extremely useful and somewhat differentiated among DeFi assets, they have significant friction for users as the price bands required to prevent them becoming over or under leveraged mean that they regularly freze and must be reset. This reset process also requires that all holders of the Synth are purged into sUSD. 
+
+The reason the price bands are required is that if the price of the underlying Synth doubles the price of the iSynth will go to 0. As the price of the iSynth tends to 0 the amount of leverage increases. For example if an ETH iSynth was instantated at $200 and ETH went to $399 then the iSynth would go to $1, so you would get $1 of price movement for each $1 spent on iETH versus $1 of price movement for every $400 spent on sETH. This property reverses in the other direction such that as the price of ETH drops you must buy significantly more iETH to get the same level of price movement. 
+
+These properties neccesitate tight price bands and increasing or removing them would add significant risk to the network. The solution proposed in this SIP therefore leaves the bands as is and instead improves the purging and reset process by incentivising anyone to call public functions to manage iSynth freezing. 
+
+## Specification
+<!--The specification should describe the syntax and semantics of any new feature, there are five sections
+1. Overview
+2. Rationale
+3. Technical Specification
+4. Test Cases
+-->
+
+# Overview
+There are three core functions required to reset an iSynth.
+1. FreezeSynth
+2. Purge
+3. ResetPrice
+
+FreeSynth - Currently the oracle is responsible for freezing the iSynth when the price limits are hit, as an aside this is also problematic for our transition to chainlink as the CL oracles do not have the ability to call freeze based on a price limit being hit. Rather than relying on a centralised service to freeze a synth we will instead open this up to be a public function that any address can call. This function will require that the current price of the underlying Synth is outside the defined price bands. If it is the synth will be frozen, if not the call will fail. The fee in SNX for freezing an iSynth will initially be set to 50 SNX. This fee is deliberately set high to ensure there are sufficient incentives for bots to operate on these Synths.
```

Comment:
We could consider handling cases where the price had reached the threshold but has since gone back below. The onus would then be to supply the synth key and the `roundId` when the price increased. This makes the keeper function a bit more useful though does mean we will need to persist the new roundId whenever a synth is reset, so that future freezes make sure the given `roundId` is above the number on reset.